### DISCUSSION TITLE
Can't roll back VDH to version that worked a few weeks ago

### DISCUSSION THREAD

**Discussion by mac-arthur1** [aclap-dev/video-downloadhelper#15] [Help and Q&A] [answered]
### Your question

VDH has worked for years, downloaded, have key etc.
After the recent requirement to use the new CoApp 
It won’t work on my Mac  10.14.6

And the roll back instructions 

If you need to rollback versions are here:
Extension: https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/
CoApp: https://www.downloadhelper.net/install-coapp-v1

Don’t work  --- never sees the CoApp
Have tried 50+ efforts to unistall , reinstall , reboot,
Did open security so it would open the “unidentified” download

Every time
Checking companion app returned: An unexpected error occurred
Even though I see it in app folder

VDH, 

Video DownloadHelper
Version 8.0.0.0
Browser locale: en-US
Production build
Built on Fri Jul 07 2023 15:02:51 GMT+0200 (Central European Summer Time)
Build options: browser=firefox

CoApp, 

Operating System, 
OS  10.14.6

Platform Mac x86-64
Browser Mozilla Firefox 115.7.0

Checking companion app returned: An unexpected error occurred

Have a key but says
The license could not be verified because the companion app is not installed

### Addon & CoApp version and details

Video DownloadHelper
Version 8.0.0.0
Browser locale: en-US
Production build
Built on Fri Jul 07 2023 15:02:51 GMT+0200 (Central European Summer Time)
Build options: browser=firefox

CoApp, 

Operating System, 
OS  10.14.6

Platform Mac x86-64
Browser Mozilla Firefox 115.7.0

Checking companion app returned: An unexpected error occurred

**Comment by paulrouget** [comment]
> Mac 10.14.6

Sadly, we do not support very old MacOS versions anymore (this version has been unsupported by Apple for 2 years now). I guess you can't update to 10.15?

Installing the CoApp V1 should have worked though. Did you install the DMG or the PKG?

**Reply by mac-arthur1** [reply]
Hi  no can't update this to 10.15. and yes tried (many times) both DMG and PKG. get the flag "CoApp is ready to use" - but VDH does not see it ?  And this worked on this laptop before I tried the failed new CoApp.  tx

**Reply by paulrouget** [reply]
Hey, I'm trying to fix your issue. Can you look at this comment: https://github.com/aclap-dev/video-downloadhelper/discussions/20#discussioncomment-8341309

**Comment by ericmarceau-rogers** [comment]
Have you looked at the earlier solution posted here?

- https://groups.google.com/g/video-downloadhelper-q-and-a/c/ZF2rDiRw8Nc/m/99lISuXsAwAJ

Eric
_just a user_

**Reply by mac-arthur1** [reply]
Don't see a solution in there - last post Jan 12 is at the point I am - trying to roll back with those two links that is what is not working for me  - I'd like to see an answer that on that Mac 10.14.6 and those links the roll back worked.  Simple not developer level. But there is not follow up, no "success !!"  tx

**Comment by GrampaWildWilly** [comment]
What was the last version of VDH for which it worked for you?  Perhaps you need to go back to a version 7.x.  Those versions are in the repository on Mozilla whose link you cited in your original post upthread here.  As for the CoApp, you can find every version there's ever been here: https://github.com/aclap-dev/vdhcoapp/releases.  You can go back to whatever version of CoApp worked for you by selecting it there.  You can choose a version 2.x or a version 1.x from there.  Now, I don't know anything about Mac.  I've never used it.  Nor Linux, which might be relevant to Mac since, as I understand it, Mac has a version of Linux inside it.  There's considerations of system-level vs user-level that I can't advise you on.  But I think it might be necessary to look at system-level & user-level things -- I can't be any more specific than "things" -- & get rid of as much of that as you can find.  I'm thinking you need to completely uninstall the browser extension & the CoApp, including all system-level & user-level things, & try again using one of the backlevel browser extensions that used to work for you & a version of the CoApp that used to work for you.  It seems that VDH 8.x does not work on the backlevel Mac operating system you are using.  In that case, you need to go back to the last version that did work for you & get a matching CoApp.

Be sure you disable automatic updates for VDH in the Firefox settings.  In the Firefox settings.  The setting for automatic update is not in the VDH settings.  VDH does not control updates of VDH.  Firefox controls that.  You must go into the Firefox settings for add-ons & disable automatic updates for VDH.  If you don't do this, you will once again get into the trouble you are in now.  You must stop automatic updates so you will be able to stay on the backlevel version of VDH.  This is a setting that does not appear until you install VDH.  So the first thing you should do after you install your chosen backlevel version of VDH is disable automatic updates for it.  Do that as quickly as you can so the browser doesn't have time to recognize that maybe it should auto-update VDH.  You should open the page about:addons before you go to install VDH.  That way, you will be able to quickly go to that page & disable automatic updates on VDH as soon as you've got it installed.  After you do that, then you can install (or not) a CoApp at your leisure.

**Reply by mac-arthur1** [reply]
**What was the last version of VDH for which it worked for you?** 
Don’t know
(I do have a time machine back up, but restoring that way would be a considerable trouble, not there yet..)
**Perhaps you need to go back to a version 7.x.** 
Did try several of those  - did not solve, but only with the 1.63 CoApp
Do have the installers for some of the earlier CoApp 
But I would like to know which pair should work with this OS  - random combinatorial trials become huge

**Those versions are in the repository on Mozilla whose link you cited in your original post upthread here. As for the CoApp, you can find every version there's ever been here: https://github.com/aclap-dev/vdhcoapp/releases.**
Yes indeed quite a list but only three are Mac.
 **But I think it might be necessary to look at system-level & user-level things -- I can't be any more specific than "things" -- & get rid of as much of that as you can find.** 
I'm thinking you need to completely uninstall the browser extension & the CoApp, including all system-level & user-level things, 
Did uninstall exten. And coapp multiple times  --- but finding “system-level & user-level things,” might be beyond me.
**It seems that VDH 8.x does not work on the backlevel Mac operating system you are using.** 
Right needs OS 15+
**last version that did work for you & get a matching CoApp.**
Right a table of which Coapps go with which extens (versions) would be helpful and even better
Which OS they need.
**Be sure you disable automatic updates for VDH in the Firefox**
Done

Thanks for a good list of thoughts and suggestions.

**Reply by mac-arthur1** [reply]
Those are good steps for stopping the auto-update  --- but I have to get back to a working VDH and Coapp combination before needing to do that, not yet  but I sure will do so when I'm there -- Thanks

**Reply by GrampaWildWilly** [reply]
I think CoApp 1.6.3 should work for you with the last 7.x version you can get from the Mozilla repository.  The 1.6.3 installers for the CoApp are on the second page of the repository.  The Mac versions don't have the word Mac in their names like the 2.x versions do.  But I gather that the files marked .dmg & .pkg are for Mac.  I assume you have one of those.  Now all we need here is for someone who actually knows Mac to come in here & say exactly what leftovers need to be removed to make it a clean install.

**Reply by GrampaWildWilly** [reply]
I just remembered something.  When CoApp 2.x came out, it put files into a new directory.  The name of the CoApp executable on Windows is vdhcoapp.exe.  That is true for both the 1.x & 2.x versions of CoApp.  If you have been messing around with both versions of the CoApp, you will have 2 executables of some name that I would expect is similar on Mac.  You need to find both of those occurrences & delete both of the directories in which you find them.  If you find only one such occurrence, delete that directory.  There's only a handful of files in those directories, & they will all be recreated in a directory that will also be created when you do finally install whatever release of CoApp you eventually choose.

**Reply by mac-arthur1** [reply]
nope, just the steps I have tried before same result
unstall exten,  trash coapp, reinstall from links

Video DownloadHelper
Version 7.6.6
Browser locale: en-US
Production build
Built on Thu Dec 08 2022 10:25:53 GMT+0100 (Central European Standard Time)
Build options: browser=firefox
Platform Mac x86-64
Browser Mozilla Firefox 115.7.0
coapp 1.6.3   says its "ready to use"

try downloading and it says
Companion application required⊗
This operation requires an external application to be completed.

Checking companion app returned: An unexpected error occurred

--- so something else is needed, needs to be done.
I'm done for tonight.   Thanks  !!

**Comment by GrampaWildWilly** [comment]
Here are the steps.  After you install VDH, you will see an entry on the about:addons page that looks like this.  You might have to reload the page to make this entry appear:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f712135e-af59-4a32-b726-dc0196ecb466)

Once you click that, a menu will appear, like this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/49ccb0e6-a914-40d6-ad57-f92964ccb429)

Clicking that menu entry will take you to another page that contains this.  Do what you see here:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d7c9e6db-a40b-464c-9e93-c6a02e5debcd)

**Reply by cybercathk** [reply]
Hi GrampaWildWilly, 

I have very a similar setup to mac-arthur1, but have downoaded the vdhcoapp-2.0.4-mac-x86_64.dmg. version & 2.0.4 helper to replace the current version which will not save anything these few months.

I cannot find the "about:addons" page in your pictures just above this post, so I'm unable to see the menu, select "Manage" or disallow & turn off automatic updates. 

Your guidance would be much appreciated.

**Reply by GrampaWildWilly** [reply]
First I don't want to oversell myself.  I do not work for VDH.  I'm just another user, like you.  Moreover, I don't know Mac or Linux.  I know only Windows.  So any advice I give here is by analogy, subject to correction by someone who actually knows Mac.  I'm a bit disappointed that nobody who knows Mac has stepped in here to help.

I'm not entirely certain but I believe none of the CoApp versions 2.x will run on old Mac releases.  @mac-arthur1 seems to have settled on VDH version 7.6.6 & CoApp 1.6.3.  That seems safely backlevel that it ought to work on older Macs.  I'm encouraged that @mac-arthur1 mac is getting the VDH settings to display his versions of the browser extension & the CoApp.  I'm assuming he's also getting it to recognize his license.  But I have no idea how to even go about investigating the error he's getting now.  Maybe you do.  You use Mac.  Do you have any ideas that might be helpful?

about:addons is Firefox-speak.  If you're not running Firefox, I don't know how to do what I described.  That's right, I also know nothing about Chrome nor Edge because I have never used either one.  The about:addons page is the way Firefox labels its page for managing add-ons.  I should hope there's something in the settings of the other browsers for doing something vaguely similar to what I show above.  Either you have to stumble around in your browser settings until you find it or you need to look up the relevant documentation.

**Reply by mac-arthur1** [reply]
too optimistic.  No does not recognize key/license

The license could not be verified because the companion app is not installed

just keeps asking to install CoApp  (to web link for 8, which won't work on less than 10.15 )

**Reply by GrampaWildWilly** [reply]
Oh.  I misinterpreted something you posted above.  So VDH is not recognizing the CoApp on the settings page.  That actually makes more sense, although it is disappointing.  And yes, that would mean your license is not recognized.

My sense is there's some leftover from the short time you had VDH 8.x & CoApp 2.x installed.  If you can find that & erase it, I am guessing that would clear your problem.  But we are still waiting for somebody who knows Mac to come in here & either explain how to do what I'm guessing needs to be done, or explain what really does need to be done.  I use Firefox on Windows & that's all I know.  Everything else I say is by analogy to this environment, & I could as easily be wrong as right in anything I say about Mac.  Where are the Mac users here?  Come on.  Speak up!

**Comment by jcfrog** [comment]
@cybercathk you need to type "about:addons" directly in the address bar, just like for a web site.

You can also try the shortcut [Command + Shift + A]

**Comment by paulrouget** [comment, marked as answer]
@mac-arthur1 please use Addon version 8 with the CoApp for old mac. See here: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation#download-for-macos-1013--1014

**Comment by cybercathk** [comment]
Hi Paul, 

Is my mac suitable?

**Reply by paulrouget** [reply]
If you use Mac, look at your MacOS version, and download the appropriate app: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation

**Comment by GrampaWildWilly** [comment]
@mac-arthur1 This looks like the start of the conversation that ended over here: https://github.com/aclap-dev/video-downloadhelper/discussions/20,  You should close this one.  I suppose you can mark this comment as the answer, since the answer is really in the other discussion.
